service-base/src/main/java/org/eclipse/hono/service/management/tenant/Tenant.java

```diff
@@ -129,7 +130,7 @@ public void setTrustedCertificateAuthority(final TrustedCertificateAuthority tru
      *
      * @param key The key of the entry.
      * @param value The value of the entry.
-     * @return This instance, to allowed chained invocations.
+     * @return This instance, to allow chained invocations.
```

Comment:
null checks?

IMHO the description should be changed to
> Adds or replaces an extension property to this tenant.

to also reflect the *put* characteristics ...

Reply:
Ok, now how about:

> Adds an extension property to this tenant. If an extension property already exist for the specified key, the old value is replaced by the specified value?

or maybe i should add a paragraph after the first sentence?

Reply:
how about extracting the `putExtension` call to a parent class and all classes requiring additional extensions will extend from this parent?

Reply:
> or maybe i should add a paragraph after the first sentence?

:+1: 

> how about extracting the putExtension call to a parent class and all classes requiring additional extensions will extend from this parent?

IMHO that is not necessary